mypy/semanal.py

```diff
@@ -7344,13 +7346,16 @@ def is_unannotated_any(typ_: Type) -> bool:
             defn.type.ret_type = NoneType()
         if self.options.infer_function_types:
             for i, arg in enumerate(defn.arguments):
+                ret = None
                 if is_unannotated_any(defn.type.arg_types[i]):
                     if arg.variable.is_inferred and arg.initializer:
                         ret = self.analyze_simple_literal_type(
                             arg.initializer, False, do_inner=True
                         )
-                        if ret:
-                            defn.type.arg_types[i] = ret
+                    if ret:
+                        defn.type.arg_types[i] = ret
+                    if arg.variable.name == "_":
```

Comment:
what if there are multiple unused arguments? since it's a syntax error to have duplicate argument names, pyright solves this problem by treating variables that are entirely underscores the same way:
```py
def foo(_, __):
    ...
```

Reply:
> pyright solves this problem by treating variables that are entirely underscores the same way

actually [no it doesn't](https://pyright-play.net/?strict=true&deprecateTypingAliases=true&enableExperimentalFeatures=true&reportCallInDefaultInitializer=true&reportImplicitStringConcatenation=true&reportImplicitOverride=true&reportMissingSuperCall=true&reportPropertyTypeMismatch=true&reportShadowedImports=true&reportUnnecessaryTypeIgnoreComment=true&reportUninitializedInstanceVariable=true&reportUnusedCallResult=true&code=CYUwZgBGD20BQH0A0EEIJQC4CwAoCBEAdCUA), at least not in this case. but i do remember seeing code somewhere in pyright that treats variables with multiple underscores the same way